packages/web/components/complex/add-conc-liquidity.tsx

```diff
@@ -0,0 +1,908 @@
+          <Image src="/icons/arrow-right.svg" height={12} width={12} />
+        </a>
+      </div>
+      <div className="flex flex-1 flex-row justify-end gap-2">
+        <PresetVolatilityCard
+          type="custom"
+          src="/images/small-vial.svg"
+          updateInputAndRangeMinMax={props.updateInputAndRangeMinMax}
+          addLiquidityConfig={props.addLiquidityConfig}
+          label="Custom"
+        />
+        <PresetVolatilityCard
+          type="passive"
+          src="/images/small-vial.svg"
+          updateInputAndRangeMinMax={props.updateInputAndRangeMinMax}
+          addLiquidityConfig={props.addLiquidityConfig}
+          label="Passive"
+        />
+        <PresetVolatilityCard
+          type="moderate"
+          src="/images/medium-vial.svg"
+          updateInputAndRangeMinMax={props.updateInputAndRangeMinMax}
+          addLiquidityConfig={props.addLiquidityConfig}
+          label="Moderate"
+        />
+        <PresetVolatilityCard
+          type="aggressive"
+          src="/images/large-vial.svg"
+          updateInputAndRangeMinMax={props.updateInputAndRangeMinMax}
+          addLiquidityConfig={props.addLiquidityConfig}
+          label="Aggressive"
+        />
+      </div>
+    </div>
+  );
+});
+
+function SelectorWrapper(props: {
+  src?: string;
+  alt?: string;
+  label?: string;
+  selected: boolean;
+  onClick: () => void;
+}) {
+  const isImage = !!props.src && !props.label;
+  const isLabel = !!props.label && !props.src;
+
+  return (
+    <div
+      className={classNames(
+        "flex h-6 cursor-pointer flex-row items-center justify-center rounded-lg bg-osmoverse-800 px-2 text-xs hover:bg-osmoverse-900",
+        {
+          "!bg-osmoverse-900": props.selected,
+        }
+      )}
+      onClick={props.onClick}
+    >
+      {isImage && (
+        <Image
+          alt={props.alt}
+          src={props.src as string}
+          width={16}
+          height={16}
+        />
+      )}
+      {isLabel && props.label}
+    </div>
+  );
+}
+
+function RangeSelector(props: {
+  label: string;
+  onClick: () => void;
+  selected: boolean;
+}) {
+  return (
+    <SelectorWrapper
+      label={props.label}
+      selected={props.selected}
+      onClick={props.onClick}
+    />
+  );
+}
+
+const DepositAmountGroup: FunctionComponent<{
+  getFiatValue?: (coin: CoinPretty) => PricePretty | undefined;
+  coin?: CoinPretty;
+  onUpdate: (amount: number) => void;
+  currentValue: Dec;
+}> = observer(({ coin, onUpdate, currentValue }) => {
+  const { priceStore, assetsStore } = useStore();
+  // const [value, setValue] = useState(0);
+
+  const { nativeBalances, ibcBalances } = assetsStore;
+
+  const fiatPer = coin?.currency.coinGeckoId
+    ? priceStore.getPrice(coin.currency.coinGeckoId, undefined)
+    : 0;
+
+  const [walletBalance] = nativeBalances
+    .concat(ibcBalances)
+    .filter((balance) => balance.balance.denom === coin?.denom);
```

Comment:
This is one way to get the balance. IMO a simplr way is to use the cosmos SDK balances query, passing in the user bech32 address and the pool denoms:

```
const balance = this._queryBalances
      .getQueryBech32Address("osmo23123123123123")
      .getBalanceFromCurrency("uosmo");
```